hardware/eHouse/EhouseEvents.cpp

```diff
@@ -0,0 +1,305 @@
+int ind=IndexOfEthDev(devh,devl);
+if (ind==0)
+    {
+    //printf("[Exec Event] RS-485, CAN, RF via PRO \r\n");
+    int mmm=0;
+    for (m=0;m<EVENT_QUEUE_MAX;m++)
+         {
+         if ((IndexOfEthDev(devh,devl)==0) && (EvQ[m]->LocalEventsTimeOuts==RUN_NOW))
+                {
+                EvQ[m]->LocalEventsTimeOuts=RE_TIME_REPLAY;      //Retry timer interval
+                memcpy(&EventBuff[EventBuffSize],EvQ[m]->LocalEventsToRun,EVENT_SIZE);   //Add to send buffer                
+        //        printf("[ via PRO ] %02x,%02x,%02x,%02x,%02x,%02x,%02x,%02x,%02x,%02x\r\n",EvQ[m].LocalEventsToRun[0],EvQ[m].LocalEventsToRun[1],EvQ[m].LocalEventsToRun[2],EvQ[m].LocalEventsToRun[3],EvQ[m].LocalEventsToRun[4],EvQ[m].LocalEventsToRun[5],EvQ[m].LocalEventsToRun[6],EvQ[m].LocalEventsToRun[7],EvQ[m].LocalEventsToRun[8],EvQ[m].LocalEventsToRun[9]);
+                EventBuffSize+=EVENT_SIZE;
+                EventsToRun[EventSize]=m;
+                EventSize++;
+                if (EventSize>=MAX_EVENTS_IN_PACK) break;       //As many events as possible
+                }
+            }
+        SendTCPEvent(EventBuff,EventSize,SrvAddrH,SrvAddrL,EventsToRun);       //Submit via tcp client to desired Ethernet eHouse controller
+        return;
+    }
+else
+    {
+      //  printf("[Exec Event] Ethernet \r\n");
+        memset(&EventBuff,0,sizeof(EventBuff));
+        EventSize=0;
+        memset(EventsToRun,0,sizeof(EventsToRun));
+        int mmm=0;
+        for (m=0;m<EVENT_QUEUE_MAX;m++)
+            {
+            if ((EvQ[m]->LocalEventsToRun[0]==devh) && (EvQ[m]->LocalEventsToRun[1]==devl) && (EvQ[m]->LocalEventsTimeOuts==RUN_NOW))
+                    {
+          //          printf("[Directly ] %02x,%02x,%02x,%02x,%02x,%02x,%02x,%02x,%02x,%02x\r\n",EvQ[m].LocalEventsToRun[0],EvQ[m].LocalEventsToRun[1],EvQ[m].LocalEventsToRun[2],EvQ[m].LocalEventsToRun[3],EvQ[m].LocalEventsToRun[4],EvQ[m].LocalEventsToRun[5],EvQ[m].LocalEventsToRun[6],EvQ[m].LocalEventsToRun[7],EvQ[m].LocalEventsToRun[8],EvQ[m].LocalEventsToRun[9]);
+                    EvQ[m]->LocalEventsTimeOuts=RE_TIME_REPLAY;								 //Retry timer interval
+                    memcpy(&EventBuff[EventBuffSize],EvQ[m]->LocalEventsToRun,EVENT_SIZE);   //Add to send buffer
+                    EventBuffSize+=EVENT_SIZE;
+                    EventsToRun[EventSize]=m;
+                    EventSize++;
+                    if (EventSize>=MAX_EVENTS_IN_PACK) break;								//As many events as possible
+                    }
+            }
+        if (devl==202)
+            {
+            m=4;
+            }
+        if (devl==201)
+            {
+            m=3;
+            }
+
+        SendTCPEvent(EventBuff,EventSize,devh,devl,EventsToRun);       //Submit via tcp client to desired Ethernet eHouse controller
+    }
+}
+/////////////////////////////////////////////////////////////////////////////////
+signed int eHouseTCP::hex2bin(const unsigned char *st,int offset)
```

Comment:
i think you can easily do this with std::stringstream ?

Reply:
Most of the code was translated (copied) from  microcontroller low-level native C. 
We need also reverse comparison of this code. If it not very neccesary we would like to keep it as similar as possible if there is such possibility